### PR TITLE
Add HIP support

### PR DESCRIPTION
**Pull request by andrerenard**
Adds AMD ROCm HIP language support. 

These files have been around for a while, but never merged because #940 was too large to easily review.

### CODE REVIEW

**Comment by andrerenard** [comment]
TODO: ~The Input/Output data command objects need to be made generic (see recent changes with the CUDA commands).~  Done now.

File: lib/hip/hipProcess.cpp
Comment (kvand) [inline lib/hip/hipProcess.cpp]: This should probably be a config option rather than a hard-uncomment, but I don't think worth holding up the merge on account of.

File: lib/hip/hipUtils.hpp
Comment (kvand) [inline lib/hip/hipUtils.hpp:22]: Should I expect this macro to be identical to the NON_OO one?

**Review by kvand — APPROVED** [review]
A couple of tiny things, but this looks good enough to me to merge in.

File: lib/hip/hipUtils.hpp
Comment (andrerenard) [reply lib/hip/hipUtils.hpp:22]: It's almost the same.  The key difference is the `NON_OO` uses the global config log level (instead of the per-stage overridable log_level).  However, in cases like this where it might be called from a function or class that doesn't inherit from `kotekanLogging` we have to use this version.

File: lib/hip/hipProcess.cpp
Comment (andrerenard) [reply lib/hip/hipProcess.cpp]: We already have a `profiling` configuration option in the base class, so that's an easy change.  However now, I'm not sure why this was commented out in the first place.  I was running profiling tests on this branch, so either this is a commit error, or we don't actually need this line.   I'll check this before merging.

Comment (andrerenard) [reply lib/hip/hipProcess.cpp]: Turns out this deals with the external profiling application, not the internal timers.  So it's simply not needed, will remove.

**Pull request merged**
